src/refactorings/convert-switch-to-if-else/convert-switch-to-if-else.ts

```diff
@@ -0,0 +1,172 @@
+  );
+}
+
+function createVisitor(
+  selection: Selection,
+  onMatch: (
+    path: t.NodePath<t.SwitchStatement>,
+    convertedNode: t.IfStatement | t.SwitchStatement
+  ) => void
+): t.Visitor {
+  return {
+    SwitchStatement(path) {
+      if (!selection.isInsidePath(path)) return;
+
+      // Since we visit nodes from parent to children, first check
+      // if a child would match the selection closer.
+      if (hasChildWhichMatchesSelection(path, selection)) return;
+
+      const convertedNode = convert(path.node);
+      if (convertedNode === path.node) return;
+
+      onMatch(path, convertedNode);
+    }
+  };
+}
+
+function hasChildWhichMatchesSelection(
+  path: t.NodePath,
+  selection: Selection
+): boolean {
+  let result = false;
+
+  path.traverse({
+    SwitchStatement(childPath) {
+      if (!selection.isInsidePath(childPath)) return;
+
+      const convertedNode = convert(childPath.node);
+      if (convertedNode === childPath.node) return;
+
+      result = true;
+      childPath.stop();
+    }
+  });
+
+  return result;
+}
+
+function convert(node: t.SwitchStatement): t.SwitchStatement | t.IfStatement {
+  try {
+    return convertNode(node);
+  } catch (err) {
+    console.log(err);
+    return node;
+  }
+}
+
+function convertNode(node: t.SwitchStatement): t.IfStatement {
+  const statements: t.Statement[] = [];
+  const fallthroughTests: t.BinaryExpression[] = [];
+
+  node.cases.forEach((caseNode, index) => {
+    const isLast = index === node.cases.length - 1;
+
+    if (!caseNode.test) {
+      if (!isLast) {
+        throw new Error("default case can only be the last case");
+      }
+
+      statements.push(caseWithoutBreak(caseNode, isLast));
+      return;
+    }
+
+    let test: t.Expression = t.binaryExpression(
+      "===",
+      node.discriminant,
+      caseNode.test
+    );
+    if (caseNode.consequent.length === 0) {
+      fallthroughTests.push(test);
+      return;
+    }
+
+    if (fallthroughTests.length > 0) {
+      fallthroughTests.reverse();
+      for (const fallthroughTest of fallthroughTests) {
+        test = t.logicalExpression("||", fallthroughTest, test);
+      }
+      fallthroughTests.splice(0, fallthroughTests.length);
+    }
+
+    const statement = t.ifStatement(test, caseWithoutBreak(caseNode, isLast));
+    statements.push(statement);
+  });
+
+  return linkIfStatements(node, statements);
+}
+
+function linkIfStatements(node: t.SwitchStatement, statements: t.Statement[]) {
+  if (statements.length === 0) {
+    return t.ifStatement(node.discriminant, t.blockStatement([]));
+  }
+
+  const firstStatement = statements[0];
+  if (firstStatement.type !== "IfStatement") {
+    throw new Error(
+      "Cannot convert switch statement with just a single default case."
+    );
+  }
+
+  if (statements.length === 1) {
+    return firstStatement;
+  }
+
+  for (let i = 0; i < statements.length - 1; i++) {
+    const statement = statements[i];
+    if (statement.type === "IfStatement") {
```

Comment:
I think we should be able to use `t.isIfStatement(statement)` instead.

Reply:
Didn't know about that, thanks